normandy/recipes/filters.py

```diff
@@ -48,6 +48,22 @@ def to_jexl(self):
         """Render this filter to a JEXL expression"""
         raise NotImplementedError
 
+    def to_operator(self, comparison):
+        if comparison == "equal":
+            return "=="
+        elif comparison == "not_equal":
+            return "!="
+        elif comparison == "greater_than":
+            return ">"
+        elif comparison == "greater_than_equal":
+            return ">="
+        elif comparison == "less_than":
+            return "<"
+        elif comparison == "less_than_equal":
+            return "<="
+        else:
+            raise serializers.ValidationError(f"Unrecognized comparison {comparison!r}")
+
 
```

Comment:
What do you think about putting a method like this on the base filter class and using when needed?

Reply:
I don't think that's the right direction. even if we were going with this kind of solution there's a couple things that seem wrong:
- the name is confusing since you're not converting the instance of the filter to an operator.
- this should likely have been a static method since it's not instance specific and is largely a utility function.

That being said I think the better approach is to extend `BaseFilter` into `BaseComparisonFilter`, something along the lines of:
```py
class BaseComparisonFilter(BaseFilter):
    value = serializers.IntegerField()
    comparison = serializers.CharField()

    @property
    def left_of_operator(self):
        raise NotImplementedError("Not correctly implemented.")

    def to_jexl(self):
        comparison = self.initial_data["comparison"]
        value = self.initial_data["value"]

        if comparison == "equal":
            operator = "=="
        elif comparison == "not_equal":
            operator = "!="
        elif comparison == "greater_than":
            operator = ">"
        elif comparison == "greater_than_equal":
            operator = ">="
        elif comparison == "less_than":
            operator = "<"
        elif comparison == "less_than_equal":
            operator = "<="
        else:
            raise serializers.ValidationError(f"Unrecognized comparison {comparison!r}")

        return f"{self.left_of_operator} {operator} {value}"
```

And then `WindowsBuildNumberFilter` would be:
```py
class WindowsBuildNumberFilter(BaseComparisonFilter):
    type = "windows_build_number"

    @property 
    def left_of_operator(self):
        return "normandy.os.windowsBuildNumber"

    @property
    def capabilities(self):
        return set()

    def to_jexl(self):
        return f"normandy.os.isWindows && ${super().to_jexl()}"
```

Reply:
ok, sounds good, thanks for writing out so much example code.